src/lib/ngsi10/UpdateContextRequest.cpp

```diff
@@ -332,16 +332,17 @@ ContextAttribute* UpdateContextRequest::attributeLookup(Entity* eP, const std::s
   {
     Entity* enP = entityVector[ceIx];
 
-    if ((enP->id != eP->id) || (enP->type != eP->type))
+    // empty type in request (enP) is always a match
+    if ((enP->id != eP->id) || ((enP->type != "") && (enP->type != eP->type)))
     {
       continue;
     }
 
-    Entity* eP = entityVector[ceIx];
```

Comment:
It was working without this change, but it is not wise to re-use as local variable the same name used in a function argument.

NCT (informative)